python/cudf/io/csv.py

```diff
@@ -235,8 +258,23 @@ def read_csv_strings(filepath, lineterminator='\n',
     csv_reader = ffi.new('csv_read_arg*')
 
     # Populate csv_reader struct
-    file_path = _wrap_string(filepath)
-    csv_reader.file_path = file_path
+    if is_file_like(filepath_or_buffer):
+        buffer = filepath_or_buffer.read()
+        # check if StringIO is used
+        if hasattr(buffer, 'encode'):
```

Comment:
If we're only checking for if the object is StringIO, BytesIO, or some other IO based Class I'd suggest we change this to check if it's an instance of the parent class IOBase instead of checking for an `encode` attribute.

https://docs.python.org/3.5/library/io.html#io.IOBase

Reply:
Maybe the main problem in this code block is the misleading comment. Namely, the intent of this check is to determine if the result of read() is str or bytes. For BytesIO, this check fails because read() returns bytes. For StringIO, the check succeeds, and we proceed to call encode to get the byte data. AFAIK, this is a good minimal way to do this check. I could also use a try except block around the encode() call, if you think that is better.

We do not really require the input buffer to be of IOBase type, as long as it has a read method and returns bytes or str. The pandas is using the same check for the input type.